packages/ngtools/webpack/src/transformers/import_factory_spec.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { tags } from '@angular-devkit/core';
+import { transformTypescript } from './ast_helpers';
+import { importFactory } from './import_factory';
+
+describe('@ngtools/webpack transformers', () => {
+  describe('import_factory', () => {
+    it('should support arrow functions', () => {
+      const input = tags.stripIndent`
+        const ɵ0 = () => import('./lazy/lazy.module').then(m => m.LazyModule);
+        const routes = [{
+          path: 'lazy',
+          loadChildren: ɵ0
+        }];
+      `;
+      const output = tags.stripIndent`
+        const ɵ0 = () => import("./lazy/lazy.module.ngfactory").then(m => m.LazyModuleNgFactory);
+        const routes = [{
+          path: 'lazy',
+          loadChildren: ɵ0
+        }];
+      `;
+
+      let warningCalled = false;
+      const transformer = importFactory(() => warningCalled = true);
+      const result = transformTypescript(input, [transformer]);
+
+      expect(tags.oneLine`${result}`).toEqual(tags.oneLine`${output}`);
+      expect(warningCalled).toBeFalsy();
+    });
+
+    it('should not transform if the format is different than expected', () => {
+      const input = tags.stripIndent`
+        const ɵ0 = function () { return import("./lazy/lazy.module").then(m => m.LazyModule); };
```

Comment:
Any reason why this is not supported, ie when targeting es5?

Reply:
Supporting this format would introduce extra complexity without much benefit. This functionality is meant to exclusively support the usecase of switching back and forth between Ivy and View Engine. It is not meant to generally support this syntax.

In regards to the target, the answer is the same as the question above.

Reply:
For reference: https://github.com/angular/angular-cli/blob/2d7022fee5418594c8d7d2c86e830f1984bfed19/packages/ngtools/webpack/src/angular_compiler_plugin.ts#L1213